content/articles/2017-08-tidyr-0.7.0.Rmd

```diff
@@ -0,0 +1,214 @@
+---
+title: tidyr 0.7.0
+slug: tidyr-0.7.0
+description: >
+  The next installment of tidyr is finally on CRAN! This version
+  brings tidy eval to a crucial component of the tidyverse workflow.
+author: Lionel Henry
+date: 2018-08-15
+photo:
+  url: https://unsplash.com/photos/8tem2WpFPhM
+  author: Radek Grzybowski
+categories: [packages]
+---
+
+```{r setup, include = FALSE}
+library("tidyr")
+library("dplyr")
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+We are happy to announce that tidyr 0.7.0 is now available on
+CRAN. This update makes tidyr compatible with the tidy evaluation
+features introduced in dplyr 0.7.0. It also uses the new tidyselect
+package as selection backend. The switch introduces some changes in
+the way selecting arguments are interpreted in order to make your
+tidyr code more robust.
+
+Install the latest version of tidyr with:
+
+```{r, eval = FALSE}
+install.packages("tidyr")
+```
+
+You can read about the complete set of changes
+at <https://github.com/tidyverse/tidyr/releases/tag/v0.7.0>.
+
+
+### New selection rules
+
+Following the switch to tidyselect, selecting functions are now
+stricter in their arguments An example of selecting function is
+`gather()` and its `...` argument. This change makes the code more
+robust by disallowing ambiguous scoping. Consider the following code:
+
+```{r}
+x <- 3
+df <- tibble(w = 1, x = 2, y = 3)
+gather(df, "variable", "value", 1:x)
+```
+
+Does it select the first three columns (using the `x` defined in the
+global environment), or does it select the first two columns (using
+the column named `x`)?
+
+To solve this ambiguity, we now make a strict distinction between data
+and context expressions. A **data expression** is either a bare name
+or an expression like `x:y` or `c(x, y)`. In a data expression, you
+can only refer to columns from the data frame. Everything else is a
+**context expression** in which you can only refer to objects that you
+have defined by assigning with `<-`.
+
+In practice this means that you can no longer refer to contextual
+objects like this:
+
+```{r, eval = FALSE}
+mtcars %>% gather(var, value, 1:ncol(mtcars))
+
+x <- 3
+mtcars %>% gather(var, value, 1:x)
+mtcars %>% gather(var, value, -(1:x))
+```
+
+You now have to be explicit about where to find objects. To do so,
+you can use the quasiquotation operator `!!` which will evaluate its
+argument early and inline the result:
+
+```{r, eval = FALSE}
+mtcars %>% gather(var, value, !! 1:ncol(mtcars))
+mtcars %>% gather(var, value, !! 1:x)
+mtcars %>% gather(var, value, !! -(1:x))
+```
+
+Read more about quasiquotation in the tidy eval section. An
+alternative is to turn your data expression into a context expression
+by using `seq()` or `seq_len()` instead of the `:` operator.
+
+```{r, eval = FALSE}
+mtcars %>% gather(var, value, seq(1, ncol(mtcars)))
```

Comment:
I think you should just skip this bit - generally I think it's better to just present one alternative in a blog post.